setup.py

```diff
@@ -64,7 +64,7 @@ def _plugin_version_trim() -> str:
         f"dbt-postgres~={_plugin_version_trim()}",
         # dbt-redshift depends deeply on this package. it does not follow SemVer, therefore there have been breaking changes in previous patch releases
         # Pin to the patch or minor version, and bump in each new minor version of dbt-redshift.
-        "redshift-connector<2.0.918,>=2.0.913,!=2.0.914",
```

Comment:
Maybe we should just update the upper bound to `<=2.1`

Reply:
@colin-rogers-dbt I pinned it in alignment with our concern of unannounced breaking changes in the `2.1.0` family if that continues to be a real concern. This is also paired with the fact this PR is behind `main`. Going with a range does open us up to that.

Is that acceptable risk?  Happy to make the adjustment if so